Factor out getCreateTime function

wagdav opened the pull request:
This patch factors out a `getCreateTime` function which tries to retrieve object creation time or the last modification time, in this order.

This PR was extracted from #21

Commit:
Factor out getCreateTime function

This patch factors out a `getCreateTime` function which tries to
retrieve object creation time or the last modification time, in this
order.

lehins commented:
This is great. Thank you.

You initial concern is also valid though, so maybe we can fix it too.

I am thinking something like this would work:
```haskell
      logAWS LevelInfo $
        "Restoring cache from " <> 
        formatRFC822 (fromMaybe createTime (resp ^. gorsLastModified)) <> 
        " with total size: " <> formatBytes len
```

Commit:
Improve the log message during restore

Display the time when the actual cache was created.  This can be either
the initial creation time of the cache object, or the last modification
date.
